### PR TITLE
Fix two navigation bugs

### PR DESCRIPTION
**Pull request by mattbowen-usds**
While using the site, I noticed small bugs in both local nav and on listing pages. This change fixes both, and also removes an unnecessary filter from gatsby-node

### CODE REVIEW

File: tech-far-hub/src/pages/template-listing.tsx [hunk 1]
```diff
@@ -47,7 +55,7 @@ const DefaultPageTemplate: React.FC<DefaultPageProps> = ({ data, children, pageC
                     </CardBody>
                     <CardFooter>
                       {!frontmatter.link && (
-                        <Link to={frontmatter.slug} className="usa-button">
+                        <Link to={`/${parent.relativeDirectory}/${frontmatter.slug}`} className="usa-button">
```
Comment (vim-usds) [inline tech-far-hub/src/pages/template-listing.tsx:58]: wow, way to get the correct link here!

File: tech-far-hub/src/pages/template-listing.tsx [hunk 2]
```diff
@@ -115,6 +123,13 @@ export const query = graphql`
           promo_description
           link
         }
+        parent {
+          ... on File {
```
Comment (vim-usds) [inline tech-far-hub/src/pages/template-listing.tsx:127]: are these Fragments? I forgot about that!

**Review by vim-usds — APPROVED** [review]
masterful!

File: tech-far-hub/src/pages/template-listing.tsx (hunk 2 above)
Comment (mattbowen-usds) [reply tech-far-hub/src/pages/template-listing.tsx:127]: I wish I could DRY up my graphql more, but these are working pretty well.

**Pull request merged**